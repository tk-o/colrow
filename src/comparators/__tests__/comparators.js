import * as comparatorsMap from '../index';

function createComparatorTest(comparator) {
  return function assertInput(input) {
    const output = input.sort(comparator);

    expect(output).toMatchSnapshot();
  }
}

function testComparator(assertInput) {
  test('sorting strings', () => {
    assertInput(['bac', 'abcd', 'aecd', '_acd', 'bbc', '', ' ', null, undefined]);
  });

  test('sorting numbers', () => {
    assertInput([1.23, -0.43, 123235235234, -Infinity, +Infinity, 0, -1234322343432, null, undefined]);
  });
}

const comparators = [...Object.values(comparatorsMap)];

comparators
  .map(createComparatorTest)
  .forEach(testComparator);
