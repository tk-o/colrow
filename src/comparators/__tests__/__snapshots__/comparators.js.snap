// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sorting numbers 1`] = `
Array [
  -Infinity,
  null,
  -1234322343432,
  -0.43,
  0,
  1.23,
  123235235234,
  Infinity,
  undefined,
]
`;

exports[`sorting numbers 2`] = `
Array [
  0,
  1.23,
  -0.43,
  Infinity,
  -Infinity,
  123235235234,
  -1234322343432,
  null,
  undefined,
]
`;

exports[`sorting numbers 3`] = `
Array [
  1.23,
  -0.43,
  123235235234,
  -Infinity,
  Infinity,
  0,
  -1234322343432,
  null,
  undefined,
]
`;

exports[`sorting strings 1`] = `
Array [
  null,
  "",
  " ",
  "_acd",
  "abcd",
  "aecd",
  "bac",
  "bbc",
  undefined,
]
`;

exports[`sorting strings 2`] = `
Array [
  null,
  "",
  " ",
  "bac",
  "bbc",
  "abcd",
  "aecd",
  "_acd",
  undefined,
]
`;

exports[`sorting strings 3`] = `
Array [
  null,
  "bac",
  "abcd",
  "aecd",
  "_acd",
  "bbc",
  "",
  " ",
  undefined,
]
`;
